dwave/plugins/torch/nn/modules/kernels.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2025 D-Wave
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Kernel functions."""
+
+from abc import abstractmethod
+
+import torch
+import torch.nn as nn
+
+from dwave.plugins.torch.nn.functional import DimensionMismatchError
+from dwave.plugins.torch.nn.modules.utils import store_config
+
+__all__ = ["Kernel", "GaussianKernel"]
+
+
+class Kernel(nn.Module):
+    """Base class for kernels.
+
+    `Kernels <https://en.wikipedia.org/wiki/Kernel_method>`_ are functions that compute a similarity
+    measure between data points. Any ``Kernel`` subclass must implement the ``_kernel`` method,
+    which computes the kernel matrix for a given input multi-dimensional tensor with shape
+    (n, f1, f2, ...), where n is the number of items and f1, f2, ... are feature dimensions, so that
+    the output is a tensor of shape (n, n) containing the pairwise kernel values.
+    """
+    @abstractmethod
```

Comment:
Cannot use abstract methods in non-abstract classes (or rather, this won't do anything).